PYME/LMVis/colourPanel.py

```diff
@@ -1,484 +1,484 @@
-        
-        #TODO - do we need the following, or can the ProcessColour module handle this internally?
-        structures = []
-        ratios = {}
-        for structure, dye in labels:
-            if structure in structures: #duplicate structure
-                structure += ' A'
-            structures.append(structure)
-            ratio = dyeRatios.getRatio(dye, mdh)
-
-            if not ratio is None:
-                ratios[structure] = float(ratio)
-
-        self.pipeline.colour_mapper.species_ratios = ratios
-        self.pipeline.Rebuild()
-
-        try:
-            self.visFr.UpdatePointColourChoices()
-            self.visFr.colourFilterPane.UpdateColourFilterChoices()
-        except AttributeError:
-            pass
-
-        self.refresh()
-
-        
-
-    def OnShow(self, event):
-        #print event.IsShown()
-        if event.IsShown() and not self.visFr.adding_panes:
-            self.refresh()
-
-    def refresh(self):
-        if not self.IsShown():
-            return
-        
-        self.colPlotPan.draw()
-        
-        self.lFluorSpecies.DeleteAllItems()
-
-        for key in self.pipeline.colour_mapper.species_ratios.keys():
-            ind = self.lFluorSpecies.InsertStringItem(UI_MAXSIZE, key)
-
-            ratio = self.pipeline.colour_mapper.species_ratios[key]
-            self.lFluorSpecies.SetStringItem(ind, 1, '%3.3f' % ratio)
-            self.lFluorSpecies.SetItemTextColour(ind, wx.Colour(*((128 * numpy.array(cm.jet_r(ratio)))[:3])))
-            
-            num_dyes = sum(self.pipeline.colourFilter._index(key))
-
-            self.lFluorSpecies.SetStringItem(ind,2, '%d' % num_dyes)
-
-
-        #self.colPlotPan._SetSize()
-        #self.colPlotPan.Refresh()
-
-
-
-
-class specAddDialog(wx.Dialog):
-    def __init__(self, parent):
-        wx.Dialog.__init__(self, parent, title='New signature')
-
-        sizer1 = wx.BoxSizer(wx.VERTICAL)
-        sizer2 = wx.BoxSizer(wx.HORIZONTAL)
-
-        sizer2.Add(wx.StaticText(self, -1, 'Name:'), 0, wx.ALIGN_CENTER_VERTICAL | wx.ALL, 5)
-
-        self.tKey = wx.TextCtrl(self, -1, value='', size=(150, -1))
-
-        sizer2.Add(self.tKey, 0, wx.ALIGN_CENTER_VERTICAL | wx.ALL, 5)
-
-        sizer2.Add(wx.StaticText(self, -1, 'Ag/(Ag + Ar)'), 0, wx.ALIGN_CENTER_VERTICAL | wx.ALL, 5)
-
-        self.tVal = wx.TextCtrl(self, -1, '.5', size=(60, -1))
-
-
-        sizer2.Add(self.tVal, 0, wx.ALIGN_CENTER_VERTICAL | wx.ALL, 5)
-
-        sizer1.Add(sizer2, 0, wx.ALIGN_CENTER_VERTICAL|wx.ALL, 5)
-
-
-        btSizer = wx.StdDialogButtonSizer()
-
-        btn = wx.Button(self, wx.ID_OK)
-        btn.SetDefault()
-
-        btSizer.AddButton(btn)
-
-        btn = wx.Button(self, wx.ID_CANCEL)
-
-        btSizer.AddButton(btn)
-
-        btSizer.Realize()
-
-        sizer1.Add(btSizer, 0, wx.ALIGN_RIGHT|wx.ALIGN_CENTER_VERTICAL|wx.ALL, 5)
-
-        self.SetSizer(sizer1)
-        sizer1.Fit(self)
-
-
+#!/usr/bin/python

```

Comment:
I'm not sure what has happened here to imply that the whole file has changed ...

Quite hard to review as is.